### PR TITLE
[8.0][ADD] product_family

### PR DESCRIPTION
**Pull request by fuentes73**
In some companies it's useful to manage and group products by families.

Usage:
1. Go to -> Sales -> Configuration -> Product families
2. Create your product families
3. Asign to your product template his product family

### CODE REVIEW

**Comment by legalsylvain** [comment]
hi @sebastienbeau. Why did you closed this PR ? 

regards.

**Comment by sebastienbeau** [comment]
Not idea, I think I have done a mistake by not closing the good one. Sorry for that

File: product_family/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Consultoría Informática Studio73 SL (contacto@studio73.es)
```
Comment (legalsylvain) [inline product_family/__init__.py:2]: minimal header just (# -*- coding: utf-8 -*-) is generaly used for init file.

File: product_family/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Consultoría Informática Studio73 SL (contacto@studio73.es)
+#          Pablo Fuentes <pablo@studio73.es>
+#
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Product Family',
+    'summary': 'Product Family',
+    'author': 'Consultoría Informática Studio 73 S.L.',
```
Comment (legalsylvain) [inline product_family/__openerp__.py:9]: Modules shared to OCA should have Odoo community Association as co author. (see other modules in the same repository for exemple). this is usefull to see easily in the appstore, which module is under OCA umbrella.

File: product_family/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Consultoría Informática Studio73 SL (contacto@studio73.es)
```
Comment (legalsylvain) [inline product_family/models/__init__.py:2]: same here.

File: product_family/models/product_family.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+class ProductFamily(models.Model):
+    _name = 'product.family'
+    _parent_name = "parent_id"
+    _order = 'parent_left'
+    _parent_order = "sequence asc, name"
+    _parent_store = True
+
+    sequence = fields.Integer(string="Sequence")
+    name = fields.Char('Nombre', required=True, translate=True)
```
Comment (legalsylvain) [inline product_family/models/product_family.py:18]: Change 'Nombre' into 'Name'.

File: product_family/models/product_family.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    parent_id = fields.Many2one(comodel_name="product.family",
+                                string="Parent Family", ondelete='cascade')
+    parent_left = fields.Integer('Left Parent', index=True)
+    parent_right = fields.Integer('Right Parent', index=True)
+    type = fields.Selection(selection=[('view', 'View'), ('normal', 'Normal')],
+                            string="Family Type", default="normal")
+
+    @api.constrains('parent_id')
+    def _check_category_recursion(self):
+        if not self._check_recursion():
```
Comment (legalsylvain) [inline product_family/models/product_family.py:28]: where is this function defined ?

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @fuentes010 . thanks for your contribution. Interesting features. Could you fix pylint errors ? (see https://travis-ci.org/OCA/product-attribute/jobs/286919410)

Some others minors remarks inline.

More general remarks. It looks that this module is quite the same as another named  product_template_tags. (V10)

https://github.com/OCA/product-attribute/pull/274

Maybe backporting this module in V8 could be better, don't you think ?

**Comment by legalsylvain** [comment]
@sebastienbeau, you closed others PR the same day (18 september) Could you check : 


- https://github.com/OCA/product-attribute/pull/214

- https://github.com/OCA/product-attribute/pull/164

- https://github.com/OCA/product-attribute/pull/127 (for once there are real reviewers ;-) )

**Comment by fuentes73** [comment]
Hi @legalsylvain thank you for your review. Yes, this module is quite similar to product_template_tags. So I think it's better backporting it to V8.
Closing this PR.

File: product_family/models/product_family.py (hunk 5 above)
Comment (fuentes73) [reply product_family/models/product_family.py:28]: Is defined at Odoo BaseModel class
https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L4924

Comment (legalsylvain) [reply product_family/models/product_family.py:28]: Thanks ! I did'nt know !

**Comment by legalsylvain** [comment]
OK, please ping me when you'll have backported this module. i'll take a look on it.
regards.

**Pull request closed without merging**